cfme/fixtures/v2v_fixtures.py

```diff
@@ -0,0 +1,531 @@
+                    pytest.skip("No host data")
+                host.update_credentials_rest(credentials=host_data[0]['credentials'])
+    except Exception:
+        logger.exception("Exception when trying to add the host credentials.")
+        pytest.skip("No data for hosts in providers, failed to retrieve hosts and add creds.")
+    # Configure conversion host for RHEV migration
+    if rhv_hosts is not None :
+        assign_conversion_tags_for_rhev(appliance, transformation_method, rhv_hosts)
+    if v2v_providers.osp_provider is not None:
+        set_conversion_instance_for_osp(appliance, transformation_method,
+                                        v2v_providers.osp_provider)
+
+
+def _tag_cleanup(host_obj, tag1, tag2):
+    """
+        Clean Up Tags
+
+        Returns: Boolean True if all Tags were removed/cleaned
+        or False means all required Tags are present on host.
+    """
+
+    def extract_tag(tag):
+        # Following strip will remove extra asterisk from tag assignment
+        return tag.category.display_name.strip(" *"), tag.display_name
+
+    valid_tags = {extract_tag(tag1), extract_tag(tag2)}
+    tags = host_obj.get_tags()
+    tags_set = set(map(extract_tag, tags))
+    # we always neeed 2 tags for migration, if total is less than 2
+    # don't bother checking what tag was it, just remove it and
+    # then add all required tags via add_tags() call. or if tags on host
+    # are not subset of valid tags, we still remove them.
+    if len(tags_set) < 2 or not tags_set.issubset(valid_tags):
+        host_obj.remove_tags(tags=tags)
+        return True
+    return False
+
+
+def create_tags(appliance, transformation_method):
+    """
+    Create tags V2V - Transformation Host * and V2V - Transformation Method
+    Args:
+        appliance:
+        transformation_method: VDDK/SSH
+
+    """
+    tag1 = appliance.collections.categories.instantiate(
+        display_name="V2V - Transformation Host *"
+    ).collections.tags.instantiate(display_name="t")
+    tag2 = appliance.collections.categories.instantiate(
+        display_name="V2V - Transformation Method"
+    ).collections.tags.instantiate(display_name=transformation_method)
+    return tag1, tag2
+
+
+def assign_conversion_tags_for_rhev(appliance, transformation_method, rhev_hosts):
+    """Assigning tags to conversion host.
+       In 5.9 all rhev hosts and tagged with tags V2V - Transformation Host=t
+       and V2V - Transformation Method=vddk/ssh.These tags are automatically removed
+       once the provider is deleted.
+
+       In 5.10 rails console commands are run to configure all the rhev hosts.
+
+    Args:
+        appliance:
+        transformation_method : vddk or ssh as per test requirement
+        rhev_hosts: number of hosts in rhev to configure for conversion
+    """
+    # Delete all prior conversion hosts otherwise it creates duplicate entries
+    appliance.ssh_client.run_rails_command("'ConversionHost.delete_all'")
```

Comment:
Should we make sure that this command passes before continuing? If this fails, we will have duplicate entries